src/App.tsx

```diff
@@ -14,11 +14,108 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+type State = {
+  selectedGoods: string[];
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  onSelect = (good: string) => {
+    this.setState(state => ({
+      selectedGoods: [...state.selectedGoods, good],
+    }));
+  };
+
+  onRemove = (good: string) => {
+    this.setState(state => ({
+      selectedGoods: state.selectedGoods.filter(item => item !== good),
+    }));
+  };
+
+  deleteAllGoods = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  };
+
+  getListToTitle = () => {
+    if (this.state.selectedGoods.length === 1) {
+      return `${this.state.selectedGoods.join(', ')} is selected`;
+    }
+
+    return `${this.state.selectedGoods.join(', ')} are selected`;
```

Comment:
try ti improve the `getListToTitle` function to print good1, good2 and lastgood , not only a list separated with coma